File: java/src/main/java/com/fleetmgr/client/sdk/event/input/user/ReleaseRejected.java
package com.fleetmgr.client.sdk.event.input.user;

/**
 * Created by: Bartosz Nawrot
 * Date: 18.09.2018
 * Description:
 */
public class ReleaseRejected extends UserEvent {

    private String message;

    public ReleaseRejected(String message) {
        super(Type.REJECT_HANDOVER);
        this.message = message;
    }

    public String getMessage() {
        return message;
    }

    @Override
    public String toString() {
        return "ReleaseRejected{" +
                "message='" + message + '\'' +
                '}';
    }
}

File: cpp/source/event/input/user/ReleaseRejected.cpp
#include "event/input/user/ReleaseRejected.hpp"

using namespace fm;
using namespace fm::event::input::user;

ReleaseRejected::ReleaseRejected() :
    UserEvent(RELEASE_REJECTED)
{
}

std::string ReleaseRejected::toString() const
{
    return "RELEASE_REJECTED: ";
}
